packages/studio-be/src/studio/cloud/cloud-client.ts

```diff
@@ -0,0 +1,125 @@
+import axios from 'axios'
+import { Logger } from 'botpress/sdk'
+import { isCDMError } from 'common/errors'
+import FormData from 'form-data'
+import { constants } from 'http2'
+
+import _ from 'lodash'
+import qs from 'qs'
+import { Result, Ok, Err } from 'ts-results'
+import { VError } from 'verror'
+import { CDMConflictError, UnexpectedError } from './errors'
+import { Bot, Introspect, OAuthAccessToken, PersonalAccessToken, Principals } from './types'
+
+export const MAX_BODY_CLOUD_BOT_SIZE = 100 * 1024 * 1024 // 100 MB
+
+export class CloudClient {
+  constructor(private logger: Logger) { }
+
+  public async createBot(props: {
+    personalAccessToken: string
+    name: string
+    workspaceId: string
```

Comment:
I really like the `props` pattern because 
* it allows the caller to be less careful about the order of arguments. For example, it's easy for the caller to mess up arguments ordering with `func(arg1: string, arg2: string, arg3: string)`, e.g. `func(arg2, arg1, arg3) // Typescript will not complain`, whereas with: `func(props: {arg1: string; arg2: string; arg3: string})` the user is less concerned about the arguments order
* If the user has 3 local variables `arg1, arg2 and arg3`, it's super easy to call the function and have Typescript help you if you mis-assign variables: `func({arg1, arg2, arg3})`, and `func({arg1, arg1, arg3}) // TS error`